db-auto-backup.py

```diff
@@ -160,70 +160,154 @@ def get_backup_provider(container_names: Iterable[str]) -> Optional[BackupProvid
-        image, tag_name = image.split(":", 1)
-        names.add(image)
+
+    if container.image.tags:
+        for tag in container.image.tags:
+            image_name = tag.split(":")[0].split("@")[0]
+            image_name = image_name.split("/")[-1]
+            names.add(image_name)
+
+    if not names and container.attrs.get("Config", {}).get("Image"):
+        image_name = container.attrs["Config"]["Image"].split(":")[0].split("@")[0]
+        image_name = image_name.split("/")[-1]
+        names.add(image_name)
+
+    if not names and container.name:
+        names.add(container.name)
+
     return names
 
+def is_swarm_mode() -> bool:
+    docker_client = docker.from_env()
+    info = docker_client.info()
+    return info.get("Swarm", {}).get("LocalNodeState") == "active"
 
-@pycron.cron(SCHEDULE)
-def backup(now: datetime) -> None:
-    print("Starting backup...")
 
+def get_local_node_id() -> str:
     docker_client = docker.from_env()
-    containers = docker_client.containers.list()
+    info = docker_client.info()
+    return info["Swarm"]["NodeID"]
 
-    backed_up_containers = []
 
-    print(f"Found {len(containers)} containers.")
+def get_local_node_tasks() -> list:
+    docker_client = docker.from_env()
+    local_node_id = get_local_node_id()
+    services = docker_client.services.list()
 
-    for container in containers:
-        container_names = get_container_names(container)
-        backup_provider = get_backup_provider(container_names)
-        if backup_provider is None:
-            continue
+    local_tasks = []
+    for service in services:
+        tasks = service.tasks()
+        for task in tasks:
+            if task["NodeID"] == local_node_id and task["Status"]["State"] == "running":
+                local_tasks.append(task)
 
-        backup_file = (
-            BACKUP_DIR
-            / f"{container.name}.{backup_provider.file_extension}{get_compressed_file_extension(COMPRESSION)}"
-        )
-        backup_temp_file_path = BACKUP_DIR / temp_backup_file_name()
+    return local_tasks
 
-        backup_command = backup_provider.backup_method(container)
-        _, output = container.exec_run(backup_command, stream=True, demux=True)
 
-        with open_file_compressed(
-            backup_temp_file_path, COMPRESSION
-        ) as backup_temp_file:
-            with tqdm.wrapattr(
-                backup_temp_file,
-                method="write",
-                desc=container.name,
-                disable=not SHOW_PROGRESS,
-            ) as f:
-                for stdout, _ in output:
-                    if stdout is None:
-                        continue
-                    f.write(stdout)
+def create_backup_file_name(container: Container, backup_provider: BackupProvider) -> Path:
+    """
+    Create a backup file name with a timestamp prefix and the container name.
+    """
+    timestamp = datetime.now().strftime('%Y%m%d_%H%M%S')
```

Comment:
Issue: This is a very big change. Backups are intentionally not timestamped. Please remove this.